Sources/Xcodeproj/pbxproj().swift

```diff
@@ -530,6 +530,29 @@ func xcodeProject(
             }
         }
     }
+    
+    // Create an aggregate target for every product for which there isn't already
+    // a target with the same name.
+    let targetNames: Set<String> = Set(modulesToTargets.values.map{ $0.name })
+    for product in graph.products {
+        // Go on to next product if we already have a target with the same name.
+        if targetNames.contains(product.name) { continue }
+        // Otherwise, create an aggreate target.
+        let target = project.addTarget(productType: nil, name: product.name)
```

Comment:
Having thought about this some more, it's pretty clear to me that we won't be able to do what we want regarding products given how Xcode's targets work.  There can be only one module per target, for example, so we will need at least one target for each Swift module.

So I think we should land this change now, given that it represents incremental improvement and that it's difficult to go further given how Xcode is currently doing things.